src/views/ObservableDetails.vue

```diff
@@ -354,8 +353,97 @@ export default {
     toggleFullscreen(fullscreen: boolean) {
       this.fullScreenEdit = !this.fullScreenEdit;
       this.editWidth = fullscreen ? "100%" : "75%";
-    }
-  },
+    },
+
+    ContextTreeView() {
+      var items = []
+      var id = 0
+      var id = 0
+      if (this.observable === null) {
+        return items
+      }
+      function treeify(root, data) {
+        if (Array.isArray(data)) {
+          var count = 0
+          for (const item of data) {
+            var element = {"id": id++, "title": count.toString(), "children": []}
+            root.push(element)
+            count++
+            treeify(element.children, item);
+          }
+        }
+        else if (typeof data === 'object') {
+          for (const [key, value] of Object.entries(data))
+            {
+              if (key === "source") {
+                continue
+              }
+              if (value === null) {
+                var title = key + ": N/A"
+                var element = {"id": id++, "title": title}
+                root.push(element)
+              }
+              else if (Array.isArray(value) || typeof value === 'object') {
+                element = {"id": id++, "title": key, "children": []}
+                root.push(element)
+                treeify(element.children, value)
+              }
+              else {
+                var title = key + ": " + value
+                var element = {"id": id++, "title": title}
+                root.push(element)
+              }
+            }
+        }
```

Comment:
This seems very similar to the function in `jsonToTreeView`, can we somehow extract them?

Reply:
done in eb701fe58a01fbc00bb64348deadbb14d205b651